ambry-api/src/main/java/com.github.ambry/network/BoundedByteBufferReceive.java

```diff
@@ -55,7 +55,6 @@ public long readFrom(ReadableByteChannel channel) throws IOException {
         sizeBuffer.flip();
```

Comment:
Minor: Should we throw some exception if `sizeBuffer.position() != sizeBuffer.capacity()`?

Reply:
Discussed this offline. The way this code is written is actually correct since it accounts for the case where the network channel does not have 8 bytes of data ready yet (but might in a subsequent call to `readFrom`)